src/PointerBox.js

```diff
@@ -3,20 +3,23 @@ import PropTypes from 'prop-types'
 import BorderBox from './BorderBox'
 import Caret from './Caret'
 import theme from './theme'
-import {Relative} from './Position'
+import {useDeprecation} from  './utils/deprecate'
 
 function PointerBox(props) {
   // don't destructure these, just grab them
-  const {bg, border, borderColor, as} = props
+  const {bg, borderColor, borderWidth, borderStyle} = props
   const {caret, children, ...boxProps} = props
-  const caretProps = {bg, borderColor, borderWidth: border, location: caret}
+  const caretProps = {bg, borderColor, borderWidth, borderStyle, location: caret}
+  useDeprecation({
+    name: 'PointerBox',
+    version: '21.0.0',
+    message: 'Use the Popover component instead. See https://primer.style/components/Popover for more details.'
+  })
   return (
-    <Relative as={as}>
```

Comment:
Removing `Relative` here fixes the issues with the caret not being positioned correctly, instead we set the relative style directly on `BorderBox`. I fixed this here even though we are deprecating this component because I needed to fix `Caret` which is being used in `Popover`.